src/fides/api/api/v1/endpoints/messaging_endpoints.py

```diff
@@ -512,21 +567,80 @@ def delete_config_by_key(
     },
 )
 def send_test_message(
+    service_type: MessagingServiceType,
+    *,
+    db: Session = Depends(deps.get_db),
+    params: MessagingTestBodyParams,
+) -> Dict[str, str]:
+    """
+    Sends a test message to the provided email or phone number.
+    """
+    config = MessagingConfig.get_by_type(db, service_type)
+    if not config:
+        raise HTTPException(
+            status_code=404,
+            detail=f"No messaging config found with service type: {service_type}",
+        )
+    try:
+        dispatch_message(
+            db,
+            action_type=MessagingActionType.TEST_MESSAGE,
+            to_identity=Identity(
+                email=params.to_identity.email,
+                phone_number=params.to_identity.phone_number,
+            ),
+            service_type=service_type.value,
+        )
+    except MessageDispatchException as exc:
+        config.update_test_status(
+            test_status=MessagingConnectionTestStatus.failed, db=db
+        )
+        raise HTTPException(
+            status_code=400,
+            detail=f"There was an error sending the test message: {exc}",
+        )
+    config.update_test_status(
+        test_status=MessagingConnectionTestStatus.succeeded, db=db
+    )
+    return {"details": "Test message successfully sent"}
+
+
+@router.post(
+    MESSAGING_TEST_DEPRECATED,
+    status_code=HTTP_200_OK,
+    dependencies=[Security(verify_oauth_client, scopes=[MESSAGING_CREATE_OR_UPDATE])],
+    response_model=Dict[str, str],
+    responses={
+        HTTP_200_OK: {
+            "content": {
+                "application/json": {
+                    "example": {
+                        "details": "Test message successfully sent",
+                    }
+                }
+            }
+        }
+    },
+)
+def send_test_message_deprecated(
     message_info: Identity,
     db: Session = Depends(deps.get_db),
     config_proxy: ConfigProxy = Depends(deps.get_config_proxy),
 ) -> Dict[str, str]:
-    """Sends a test message."""
+    """
+    This endpoint is deprecated. Please use the `POST /messaging/test/{service_type}`
+    """
     try:
         dispatch_message(
             db,
             action_type=MessagingActionType.TEST_MESSAGE,
             to_identity=message_info,
             service_type=config_proxy.notifications.notification_service_type,
         )
-    except MessageDispatchException as e:
+    except MessageDispatchException as exc:
         raise HTTPException(
-            status_code=400, detail=f"There was an error sending the test message: {e}"
+            status_code=400,
+            detail=f"There was an error sending the test message: {exc}",
         )
     return {"details": "Test message successfully sent"}
```

Comment:
You can also add a deprecation header so anyone using but not commonly looking at the docs will get a notice. We could even add a date like however many mos in the future it will be fully deprecated so we can remove it. 

```suggestion
    return {
        "details": "Test message successfully sent", 
        "warnings": ["This endpoint is deprecated. Please use the `POST /messaging/test/{service_type}`"]
    }
```